azurerm/data_source_public_ip_prefix.go

```diff
@@ -0,0 +1,65 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmPublicIpPrefix() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPublicIpPrefixRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"prefix_length": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"ip_prefix": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"zones": azure.SchemaSingleZone(),
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func dataSourceArmPublicIpPrefixRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).network.PublicIPPrefixesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	resp, err := client.Get(ctx, resGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Public IP prefix %q was not found", name)
+		}
+		return err
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmPublicIpPrefixRead(d, meta)
```

Comment:
in-case the Schema's diverge over time - rather than calling out to the Resource's Read Function could we instead set these inline here?

```suggestion
	d.Set("zones", resp.Zones)
	if location := resp.Location; location != nil {
		d.Set("location", azure.NormalizeLocation(*location))
	}

	if sku := resp.Sku; sku != nil {
		d.Set("sku", string(sku.Name))
	}

	if props := resp.PublicIPPrefixPropertiesFormat; props != nil {
		d.Set("prefix_length", props.PrefixLength)
		d.Set("ip_prefix", props.IPPrefix)
	}

	return tags.FlattenAndSet(d, resp.Tags)
```

Reply:
Inline makes sense.

I was curious why we wouldn't add the [`name` and `resource group` lines](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_public_ip_prefix.go#L138-L139) here as well?